pages/conditional-includes.md

```diff
@@ -0,0 +1,92 @@
+/>
+
+# Keeping up with two git identities
+
+_November 13, 2018_
+
+If you are working in the frontend, then you most likely use a lot of open-source projects and perhaps your company even open sourced several internal projects. Also you most likely fix issues you have with 3rd party projects you depend on from time to time.
+
+And everything works fine with you simple global git name and email, but then you start to notice some of your GitHub commits lack an avatar, or quite the opposite in your internal source control system (e.g. hosted BitBucket or Gitlab) your commits lack the avatar and username link is shaded.
+
+You start to wonder why and after some digging you will find out one of two: either you have been committing to open-source project with your company email or to internal projects with your private email. None of these two options are good.
+
+Dont worry, because with git [2.13](https://github.com/git/git/blob/v2.13.0/Documentation/RelNotes/2.13.0.txt) you get a feature called [conditional includes](https://git-scm.com/docs/git-config#_includes)
+
+It allows you to configure git to take file system into consideration to figure out which credentials to use for commit.
+
+Say you keep your repositories in a specific folder or folders. And structure of those folder probably look similar to this:
+
+```
+☯ tree projects/ -L 2
+projects/
+├── company
+│   ├── route
+│   ├── scripts
+│   └── tools
+└── oss
+    ├── iamstarkov.com
+    ├── mdx
+    └── next.js
+```
+
+Then you want to use different identities for different purposes. Conditional includes will help with that.
+
+> You can include a config file from another conditionally by setting a `includeIf.<condition>.path` variable to the name of the file to be included.  
+> [git docs](https://git-scm.com/docs/git-config#_conditional_includes)
+
+In order to utilise this feature you will need three files:
+
+- `~/.gitconfig`
+- `~/.gitconfig_oss`
+- `~/.gitconfig_company`
+
+And content of these files will look like this:
+
+```
+$ cat ~/.gitconfig
+[user]
+  name = Vladimir Starkov
+[includeIf "gitdir:~/projects/oss/"]
+    path = ~/.gitconfig_oss
+[includeIf "gitdir:~/projects/company/"]
+    path = ~/.gitconfig_company
+
+$ cat ~/.gitconfig_oss
+[user]
+  email = iamstarkov@gmail.com
+
+$ cat ~/.gitconfig_company
+[user]
+  email = Vladimir.Starkov@company.org
+```
+
+Let's go through what is going on here.
+
+in `~/.gitconfig` you specify general options which can be overridden by subconfigs. In this case you specify `user.name` option and it is `Vladimir Starkov` in my case. After that you have two `includeIf` sections, with `gitdir` option set to relevant folders with paths set to relevant subconfigs.
+Which in turn specify different `user.email` values for different purposes: in my case it is `user.email=iamstarkov@gmail.com` for open-source projects and `user.email=Vladimir.Starkov@company.org`. It was fast, huh?
```

Comment:
maybe use bullet-points?
---
* `user.email=iamstarkov@gmail.com` for open-source
* `user.email=Vladimir.Starkov@company.org`  for company.org projects

Reply:
nah